jax/experimental/jax2tf/tests/jax2tf_test.py

```diff
@@ -410,6 +415,238 @@ def outer(y):
+
+  def test_pytree(self):
+    """Arguments and in_shapes are pytrees."""
+
+    # Arguments are of the form [([x00, x01], [x10]), dict(a=ya, b=yb)]
+    def add_all_jax(x_pair_of_list, y_dict):
+      x_list_0, x_list_1 = x_pair_of_list
+      return functools.reduce(operator.add,
+                              x_list_0 + x_list_1 + [y_dict["a"], y_dict["b"]])
+
+    self.CheckShapePolymorphism(add_all_jax, [(["(v, 4)", "(v, 4)"], [("v, 4")]),
+                                              dict(a="(v, 4)", b="(v, 4)")],
+                                expected_output_shape="(v, 4)")
+    # Now partial in_shapes; the parts of the in_shapes that are not specified
+    # must have full input_signatures.
+    full_signature = self.MakeInputSignature((["(8)", "(8,)"], [("8,")]),
+                                             dict(a="(8,)", b="(8,)"))
+    self.CheckShapePolymorphism(add_all_jax, None,
+                                input_signature=full_signature,
+                                expected_output_shape="(8,)")
+
+    self.CheckShapePolymorphism(add_all_jax, [None, None],
+                                input_signature=full_signature,
+                                expected_output_shape="(8,)")
+    self.CheckShapePolymorphism(add_all_jax, [(None, None), None],
+                                input_signature=full_signature,
+                                expected_output_shape="(8,)")
+    self.CheckShapePolymorphism(add_all_jax, [(None, [("v, 4")]),
+                                              dict(a="(v, 4)", b=None)],
+                                input_signature=self.MakeInputSignature((["(4,)", "(4,)"], [("v, 4,")]),
+                                                                        dict(a="(v, 4)", b="(4,)")),
+                                expected_output_shape="(v, 4)")
+
+  def test_with_custom_vjp(self):
+    """Shape-polymorphic custom VJP."""
+    # TODO: is this test really adding anything???
+    @jax.custom_vjp
+    def f(x):
+      # x: [b1, b2, d1, d2] (a batch of matrices)
+      # res: [b1, b2, d1, d1]
+      return jnp.matmul(x, jnp.transpose(x, axes=(0, 1, 3, 2)))
+
+    # f_fwd: a -> (b, residual)
+    def f_fwd(x):
+      # x: [b1, b2, d1, d2]
+      # b: [b1, b2, d1, d1]
+      # residual: [b1, b2, d1, d2]
+      return f(x), 3. * x
+    # f_bwd: (residual, CT b) -> [CT a]
+    def f_bwd(residual, ct_b):
+      # residual: [b1, b2, d1, d2]
+      # ct_b: [b1, b2, d1, d1]
+      # ct_a: [b1, b2, d1, d2]
+      return jnp.matmul(ct_b, residual),
+
+    f.defvjp(f_fwd, f_bwd)
+    x = np.ones((2, 3, 4, 5), dtype=np.float32)
+    res_jax = f(x)
+    res_jax_grad = jax.grad(lambda x: jnp.sum(f(x)))(x)
+
+    f_tf = self.CheckShapePolymorphism(f, ["(batch1, batch2, d1, d2)"],
+                                       expected_output_shape="(batch1, batch2, d1, d1)")
+    self.assertAllClose(res_jax, f_tf(x))
+
+    xv = tf.Variable(x, dtype=dtypes.canonicalize_dtype(jnp.float_))
```

Comment:
As it is, it will fail in x64 mode (same as below). The line should be:
```suggestion
    xv = tf.Variable(x, dtype=jnp.float32)
```

Reply:
Fixed